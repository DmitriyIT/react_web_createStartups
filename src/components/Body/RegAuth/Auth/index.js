import React from 'react';
import './Auth.scss';

function Auth() {
	return (
		<div className="AuthReg">
			<div className="AuthReg__title">email</div>
			<div className="AuthReg__field">pass</div>
			<div className="AuthReg__swith">Регистрация</div>
			<div className="AuthReg__button">Войти</div>
		</div>
	);
}

export default Auth;