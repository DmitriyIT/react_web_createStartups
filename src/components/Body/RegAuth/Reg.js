import React from 'react';
import { Link } from 'react-router-dom';

function Reg({ switch2, onChangeEmail }) {
	return (
		<div className="AuthReg">
			<div className="AuthReg__title">Регистрация</div>
			<input className="AuthReg__field" placeholder="введите ваш email" onChange={onChangeEmail} />
			<input className="AuthReg__field" placeholder="password" />
			<input className="AuthReg__field" placeholder="password еще раз" />
			<div className="AuthReg__swith" onClick={switch2}>Регистрация</div>	
			<Link to="">
				<div className="AuthReg__button">Зарегистрироваться</div>
			</Link>
		</div>
	);
}

export default Reg;