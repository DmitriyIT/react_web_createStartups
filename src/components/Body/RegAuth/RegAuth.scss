.AuthReg {
	width: 430px;
	border-radius: 10px 10px 5px 5px;
	overflow: hidden;
	box-shadow: 1px 4px 20px rgba(0, 0, 0, 0.15);

	font-family: Fira Sans;
	font-style: italic;
	font-weight: 500;
	font-size: 14px;
	padding-left: 28px;
	padding-bottom: 28px;

	.AuthReg__title {
		height: 55px;
		margin-left: -28px;
		padding-left: 18px;
		margin-bottom: 22px;
		background: linear-gradient(162.61deg, #FFE99C 16.29%, #FFE486 115.92%, rgba(255, 164, 91, 0.8) 166.56%);
		font-size: 16px;
		line-height: 55px;
		color: #24505F;
	}
	.AuthReg__err {
		color: rgba(207, 88, 88, 1);
	}
	.AuthReg__field {
		display: block;
		width: 70%;
		height: 35px;
		margin-bottom: 22px;
		padding-left: 10px;
		background: #F3F3F3;
		border-radius: 5px;
		border: none;
		outline: none;

		font-family: Fira Sans;
		font-style: italic;
		font-weight: 500;
		font-size: 14px;
		line-height: 35px;
		color: #345761;
	}
	.AuthReg__swith {
		display: inline-block;
		line-height: normal;
		color: #3AA5C4;
	}
	.AuthReg__button {
		display: inline-block;
		min-width: 160px;
		height: 40px;
		margin-left: 50px;
		border: none;
		text-align: left;
		background: linear-gradient(156.98deg, #FFE486 145.71%, #009D90 431.65%);
		font-size: 16px;
		font-weight: 400;
		line-height: 38px;
		color: #265D6E;
		padding-left: 15px;
		padding-right: 15px;
	}
}