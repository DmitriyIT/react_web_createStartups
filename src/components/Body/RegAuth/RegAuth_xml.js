import React from 'react';
import { Link } from 'react-router-dom';

function AuthReg({ submitForm, switch2, onChangeEmail, isSwitch, err, comment }) {
	var title = isSwitch ? 'Авторизация' : 'Регистрация';
	var button = isSwitch ? 'Войти' : 'Зарегистрироваться';
	var switchA = isSwitch ? 'Регистрация' : 'Авторизация';

	return (
		<div className="AuthReg">
			<div className="AuthReg__title">{title}</div>
			{ comment ? <div className="AuthReg__comment">{comment}</div> : ''}
			{ err ? <div className="AuthReg__err">{err}</div> : ''}
			<form onSubmit={submitForm}>
				<input name="email" className="AuthReg__field" placeholder="email" onChange={onChangeEmail}></input>
				<input name="password" className="AuthReg__field" placeholder="пароль"></input>
				
				{!isSwitch ? <input name="password2" className="AuthReg__field" placeholder="пароль еще раз"></input> : ''}
				<div className="AuthReg__swith" onClick={switch2}> {switchA}</div>	
				<input type="submit" className="AuthReg__button" value={button} />
			</form>
		</div>
	);
}

export default AuthReg;