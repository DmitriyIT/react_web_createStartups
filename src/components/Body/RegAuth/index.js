import React, { Component } from 'react';
import './RegAuth.scss';

import AuthReg from './AuthReg.js';

class RegAuth extends Component {
	constructor(props) {
		super(props);
		this.state = {
			auth_show: true,
			err: ''
		};
	}

	goSwitch = () => {
		this.setState((state, props) => ({
			auth_show: !state.auth_show 
		}));
	}

	onChangeEmail = (e) => {
		var email = e.target.value;
		
		(this.state.auth_show == false) ? 
			this.checkEmail(email)
			: this.setState({err: ''});

		this.setState({email: email});
	}

	checkEmail = (email) => {
		fetch("/checkEmail", {
		    headers: {
		      'Accept': 'application/json',
		      'Content-Type': 'application/json'
		    },
		    method: "POST",
		    body: JSON.stringify({email: email})
		})
		.then((response) => response.json())
		.then((data) => { 
			console.log(data.ans)
			data.ans ?
				this.setState({err: 'такой email уже используется'})
				: this.setState({err: ''});
		})
		.catch(res => console.log('catch' + res));
	}

	refForm = React.createRef();

	submitForm = (e) => {
		e.preventDefault();

		var email     = e.target.email.value,
			 password  = e.target.password.value;

		if (this.state.auth_show) {
			// Auth
			this.makeAuth(email, password);

		} else {
			// Registration
			var password2 = e.target.password2.value;
			if (password === password2) {
				this.makeReg(email, password);
				// this.setState({err: ''});
			} else {
				this.setState({err: 'повторно введенный пароль отличается'});
			}
		}
	}

	makeAuth = (email, password) => {
		fetch("/login", {
		    headers: {
		      'Accept': 'application/json',
		      'Content-Type': 'application/json'
		    },
		    method: "POST",
		    body: JSON.stringify({ email: email, password: password})
		})
		.then((response) => response.json())
		.then((data) => {
			// var response={
			// 	code:int,
			// 	userData:{
			// 		name:"str",
			// 		surName:"str",
			// 		photo:"str",
			// 		description:"",
			// 		notifications:[],
			// 		myStartupId:int
			// 	}
			// }
			
			// console.log(data.code + ' userData.name: ' + data.userData.name);
			if (data) {
				window.location.assign('/');
			} else {
				this.setState({err: 'введен неверный email или пароль'});
			}
		})
		.catch(function(res){ console.log('catch' + res) });
	}

	makeReg = (email, password) => {
		fetch("/reg", {
		    headers: {
		      'Accept': 'application/json',
		      'Content-Type': 'application/json'
		    },
		    method: "POST",
		    body: JSON.stringify({ email: email, password: password})
		})
		.then((response) => response.json())
		.then((data) => {
			// var response={
			// 	code:int,
			// 	userData:{
			// 		name:"str",
			// 		surName:"str",
			// 		photo:"str",
			// 		description:"",
			// 		notifications:[],
			// 		myStartupId:int
			// 	}
			// }
			
			// console.log(data.code + ' userData.name: ' + data.userData.name);
			if (data) {
				window.location.assign('/');
			} else {
				this.setState({err: 'ошибка сервера, приносим извинения'});
			}
		})
		.catch(function(res){ console.log('catch: ' + res) });
	}
	render() {
		return (
			<AuthReg 
				submitForm={this.submitForm} 
				switch2={this.goSwitch} 
				onChangeEmail={this.onChangeEmail} 
				isSwitch={this.state.auth_show} 
				err={this.state.err} />
		);
	}
}

export default RegAuth;