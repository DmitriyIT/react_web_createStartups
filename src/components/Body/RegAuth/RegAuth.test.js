import React, { Component } from 'react';
import { Link, BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import RegAuth from './index.js';

var props = {
	linkHappyPath: '/profile',
	comment: 'чтобы войти в профиль нужно зарегистрироваться'
}

it('base path, all include', () => {
	var component = renderer.create(
		<BrowserRouter>
			<RegAuth {...props} />
		</BrowserRouter>
	);
	expect(component.toJSON()).toMatchSnapshot();
})

it('without comment', () => {
	delete(props.comment);
	var component = renderer.create(
		<BrowserRouter>
			<RegAuth {...props} />
		</BrowserRouter>
	);
	expect(component.toJSON()).toMatchSnapshot();
})
