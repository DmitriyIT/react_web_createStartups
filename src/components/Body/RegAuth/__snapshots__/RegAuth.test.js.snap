// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base path, all include 1`] = `
<div
  className="AuthReg"
>
  <div
    className="AuthReg__title"
  >
    Авторизация
  </div>
  <div
    className="AuthReg__comment"
  >
    чтобы войти в профиль нужно зарегистрироваться
  </div>
  
  <form
    onSubmit={[Function]}
  >
    <input
      className="AuthReg__field"
      name="email"
      onChange={[Function]}
      placeholder="email"
    />
    <input
      className="AuthReg__field"
      name="password"
      placeholder="пароль"
    />
    
    <div
      className="AuthReg__swith"
      onClick={[Function]}
    >
       
      Регистрация
    </div>
    <input
      className="AuthReg__button"
      type="submit"
      value="Войти"
    />
  </form>
</div>
`;

exports[`without comment 1`] = `
<div
  className="AuthReg"
>
  <div
    className="AuthReg__title"
  >
    Авторизация
  </div>
  
  
  <form
    onSubmit={[Function]}
  >
    <input
      className="AuthReg__field"
      name="email"
      onChange={[Function]}
      placeholder="email"
    />
    <input
      className="AuthReg__field"
      name="password"
      placeholder="пароль"
    />
    
    <div
      className="AuthReg__swith"
      onClick={[Function]}
    >
       
      Регистрация
    </div>
    <input
      className="AuthReg__button"
      type="submit"
      value="Войти"
    />
  </form>
</div>
`;
